src/java/org/apache/cassandra/index/sai/disk/v1/SSTableIndexWriter.java

```diff
@@ -213,18 +219,53 @@ else if (shouldFlush(sstableRowId))
             currentBuilder = newSegmentBuilder();
         }
 
-        if (term.remaining() == 0) return;
+        if (term.remaining() == 0)
+            return;
 
-        if (!TypeUtil.isLiteral(type))
+        updatesInFlight.incrementAndGet();
+        ForkJoinPool.commonPool().submit(() -> {
+            try
+            {
+                addTermInternal(term, key, sstableRowId, type);
+            }
+            finally
+            {
+                updatesInFlight.decrementAndGet();
+            }
+        });
+        busyWait(() -> termSizeReservoir.size() == 0);
```

Comment:
Readability nit: Consider renaming to `busyWaitWhile`. Then I wouldn't need to check the implementation to learn if it is `while` or `until`.